### PR TITLE
Fix flaky smoketest due to race condition in subscribe

### PR DESCRIPTION
**Pull request by coolreader18**
This should fix it - now it'll only error if the initial update was never printed, i.e. the subscription never started. I'm about 90% sure that this should fix any race condition issues, but if there's any flakiness we can just go back to the fix from before.

### CODE REVIEW

File: smoketests/tests/add_remove_index.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def test_add_then_remove_index(self):
         # There are no indices, resulting in an unsupported unindexed join.
         self.publish_module(name, clear = False)
         with self.assertRaises(Exception):
-            self.subscribe(self.JOIN_QUERY, n = 0)()
+            self.subscribe(self.JOIN_QUERY, n = 0)
```
Comment (bfops) [inline smoketests/tests/add_remove_index.py:75]: if we've fixed the race condition, this change from #1536 is no longer necessary

File: smoketests/__init__.py [hunk 2]
```diff
@@ -237,11 +237,14 @@ def stderr_task():
...
-            raise subprocess.CalledProcessError(code, fake_args)
-        print("initial update:", init_update)
+        init_update = proc.stdout.readline().strip()
+        if init_update:
+            print("initial update:", init_update)
+        else:
+            code = proc.poll()
+            if code:
+                raise subprocess.CalledProcessError(code, fake_args)
+            print("no inital update, but no error code either")
```
Comment (bfops) [inline smoketests/__init__.py:247]: to double-check - is it intended that we trigger this _both_ if the `code` is `None` _and_` if the `code` is `0`?

Comment (coolreader18) [reply smoketests/__init__.py:247]: Correct. Maybe it should check for 0, but it doesn't make much of a difference anyway.

**Review by bfops — APPROVED** [review]
Thanks for looking into this! Seems worth trying. It's definitely nicer UX if it's free of race conditions.

File: smoketests/__init__.py (hunk 2 above)
Comment (coolreader18) [reply smoketests/__init__.py:247]: Oh - it should wait. I see the potential race condition here.

Comment (bfops) [reply smoketests/__init__.py:247]: Yeah I think that's right. I suspect it's basically when the process hasn't quite exited yet, but I was unable to reproduce locally so I couldn't really narrow done.

**Comment by coolreader18** [comment]
Ran it `test_add_then_remove_index` 20 times with no failure, so it seems fine now, hopefully.

**Review by bfops — APPROVED** [review]
LGTM let's give it a shot

**Pull request merged**